auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailFragment.java

```diff
@@ -98,6 +111,9 @@ protected void onFailure(@NonNull Exception e) {
                             R.integer.fui_min_password_length));
                 } else if (e instanceof FirebaseAuthInvalidCredentialsException) {
                     mEmailInput.setError(getString(R.string.fui_invalid_email_address));
+                } else if (e instanceof FirebaseAuthAnonymousUpgradeException) {
+                    IdpResponse response = ((FirebaseAuthAnonymousUpgradeException) e).getResponse();
+                    ((AnonymousUpgradeListener) getActivity()).onMergeFailure(response);
```

Comment:
This is fine, however you could optionally assert the cast in `onActivityCreated` and then store a listener reference rather than casting here.